plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/reflection/HtmlReflectionImpl.java

```diff
@@ -151,218 +205,25 @@ public static boolean isChildOf(Class<?> parent, Field field) {
-            throw new FactoryRuntimeException(String.format("Internal error during attempt to find block '%s'", blockPath), ex);
-        }
-    }
-
-    /**
-     * See {@link HtmlReflectionUtils#findBlocks(String, Object, boolean)} for description.
-     * Search is being performed on a current page
+     * Check if corresponding field is a block. I.e. it has
+     * {@link ElementTitle} annotation and extends {@link HtmlElement} class
+     * directly
      *
-     * @param page the page on which the method will be executed
-     * @param blockPath full path or just a name of the block to search
-     * @return list of objects that were found by specified path
+     * @param field field that is being checked
+     * @return true|false
      */
-    public static List<HtmlElement> findBlocks(Page page, String blockPath) throws NoSuchElementException {
-        try {
-            return findBlocks(blockPath, page, false);
-        } catch (IllegalAccessException ex) {
-            throw new FactoryRuntimeException(String.format("Internal error during attempt to find a block '%s'", blockPath), ex);
-        }
+    private boolean isBlockElement(Field field) {
+        return (null != field.getAnnotation(ElementTitle.class))
+                && isChildOf(HtmlElement.class, field);
     }
 
-    /**
-     * Execute parameter-less method inside of the given block element.
-     *
-     * @param page the page on which the method will be executed
-     * @param block block title, or a block chain string separated with '-&gt;'
-     * symbols
-     * @param actionTitle title of the action to execute
-     * @throws NoSuchMethodException if required method couldn't be
-     * found
-     */
-    public static void executeMethodByTitleInBlock(Page page, String block, String actionTitle) throws NoSuchMethodException {
+    @Override
+    public void executeMethodByTitleInBlock(Page page, String block, String actionTitle) throws NoSuchMethodException {
         executeMethodByTitleInBlock(page, block, actionTitle, new Object[0]);
     }
 
-    /**
-     * Execute method with one or more parameters inside of the given block
-     * element !BEWARE! If there are several elements found by specified block
-     * path, a first one will be used!
-     *
-     * @param page the page on which the method will be executed
-     * @param blockPath block title, or a block chain string separated with
-     * '-&gt;' symbols
-     * @param actionTitle title of the action to execute
-     * @param parameters parameters that will be passed to method
-     * @throws NoSuchMethodException if required method couldn't be
-     * found
-     */
-    public static void executeMethodByTitleInBlock(Page page, String blockPath, String actionTitle, Object... parameters) throws NoSuchMethodException {
+    @Override
+    public void executeMethodByTitleInBlock(Page page, String blockPath, String actionTitle, Object... parameters) throws NoSuchMethodException {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. ([squid:S3776](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS3776))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=101&issues=AWbDPCupSgN4uCclZPgc&open=AWbDPCupSgN4uCclZPgc)

[ISSUE_KEY]: AWbDPCupSgN4uCclZPgc